radix-transactions/src/manifest/static_resource_movements/typed_invocation.rs

```diff
@@ -34,74 +33,91 @@ macro_rules! define_typed_invocations {
+            #[derive(Debug)]
             pub enum TypedNativeInvocation {
                 $(
                     [< $package_name Package >]([< $package_name Invocations >])
                 ),*
             }
 
             impl TypedNativeInvocation {
-                pub fn from_method_invocation(
-                    address: &NodeId,
-                    module_id: ::radix_engine_interface::prelude::ModuleId,
+                pub fn from_main_module_method_invocation(
+                    address: &GlobalAddress,
                     method_name: &str,
                     args: &::radix_common::prelude::ManifestValue
-                ) -> Option<Self> {
-                    match (address.entity_type(), module_id) {
+                ) -> Result<Option<Self>, StaticResourceMovementsError> {
+                    Ok(match address.as_node_id().entity_type() {
                         $(
                             $(
-                                (Some($entity_type_pat), $module_id) => {
-                                    [< $blueprint_name Method >]
-                                        ::from_invocation(
-                                            method_name,
-                                            args
+                                Some($entity_type_pat) => {
+                                    let invocation = [< $blueprint_name Method >]::from_invocation(method_name, args)?;
+                                    Some(Self::[< $package_name Package >](
+                                        [< $package_name Invocations >]::[< $blueprint_name Blueprint >](
+                                            [< $blueprint_name BlueprintInvocations >]::Method(invocation)
                                         )
-                                        .and_then(|invocation| $type::try_from(address.as_bytes()).ok().map(|address| (address, invocation)))
-                                        .map(|(address, invocation)|
-                                            [< $blueprint_name BlueprintInvocations >]::Method(address, invocation)
-                                        )
-                                        .map([< $package_name Invocations >]::[< $blueprint_name Blueprint >])
-                                        .map(Self::[< $package_name Package >])
-
+                                    ))
                                 }
                             )*
                         )*
                         _ => None
-                    }
+                    })
+                }
+
+                pub fn from_blueprint_method_invocation(
+                    package_address: &PackageAddress,
+                    blueprint_name: &str,
+                    method_name: &str,
+                    args: &::radix_common::prelude::ManifestValue
+                ) -> Result<Option<Self>, StaticResourceMovementsError> {
+                    Ok(match *package_address {
```

Comment:
This doesn't work.

We need a different error message for "package address not matched" (the package address is not native - or not supported yet - so we return None for not-matched)" to "package address is native and method is not detected" (this is an error, because our method mapping for each package should be complete)